tests/Unit/Domain/Test_Neighbors.cpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "Domain/Neighbors.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+TEST_CASE("Unit.Domain.Neighbors", "[Domain][Unit]") {
+  // Test default constructor:
+  Neighbors<3> test_neighbors;
+  static_cast<void>(test_neighbors);
+
+  // Test constructor:
+  const Orientation<3> custom_orientation(std::array<Direction<3>, 3>{
+      {Direction<3>::upper_eta(), Direction<3>::upper_zeta(),
+       Direction<3>::upper_xi()}});
+
+  std::unordered_set<ElementId<3>> custom_ids;
+  std::array<SegmentId, 3> segment1_ids
+      {{SegmentId(2, 3), SegmentId(1, 0), SegmentId(1, 1)}};
+  ElementId<3> element1_id(2, segment1_ids);
+  custom_ids.insert(element1_id);
+
+  std::array<SegmentId, 3> segment2_ids
+      {{SegmentId(2, 2), SegmentId(1, 1), SegmentId(1, 0)}};
+  ElementId<3> element2_id(3, segment2_ids);
+  custom_ids.insert(element2_id);
+
+  std::array<SegmentId, 3> segment3_ids
+      {{SegmentId(2, 1), SegmentId(1, 0), SegmentId(1, 1)}};
+  ElementId<3> element3_id(1, segment3_ids);
+  custom_ids.insert(element3_id);
+
+  Neighbors<3> custom_neighbors(custom_ids, custom_orientation);
+
+  // Test size
+  CHECK(custom_neighbors.size() == 3);
+
+  // Test output
+  CHECK(get_output(custom_neighbors) ==
```

Comment:
remove the output checks for now until we decide about deterministic printing of unordered containers